File: apps/admin-v2/src/infrastructure/api/mappers/user.mapper.ts
import type { User } from '@/domain/entities/user';

export function mapUser(raw: Record<string, unknown>): User {
  return {
    id: raw.id as string,
    name: raw.name as string,
    email: (raw.email as string | null) ?? null,
    username: (raw.username as string | null) ?? null,
    phone: (raw.phone as string) ?? null,
    isSuperAdmin: (raw.is_super_admin as boolean) ?? false,
    createdAt: raw.created_at ? new Date(raw.created_at as string) : new Date(),
    role: raw.role as User['role'],
  };
}

File: apps/admin_mobile/lib/infrastructure/api/mappers/user_mapper.dart
import '../../../domain/entities/user.dart';

User mapUser(Map<String, dynamic> json) {
  return User(
    id: json['id'] as int,
    name: json['name'] as String,
    email: json['email'] as String,
    phone: json['phone'] as String?,
    isSuperAdmin: json['is_super_admin'] as bool? ?? false,
    createdAt: DateTime.parse(json['created_at'] as String),
    role: json['role'] != null ? UserRole.fromApi(json['role'] as String) : null,
  );
}
